### PR TITLE
fix(haiku): execve not allowed if other pthreads have already been created

### PR DESCRIPTION
**Pull request by Sylvain78**
Compiling for Haiku

### CODE REVIEW

File: otherlibs/stdune/src/proc.ml [hunk 1]
```diff
@@ -11,7 +11,7 @@ let restore_cwd_and_execve prog argv ~env =
   (* run at_exit before changing the working directory *)
   Stdlib.do_at_exit ();
   Sys.chdir (Path.External.to_string Path.External.initial_cwd);
-  if Sys.win32
+  if Sys.win32 || Platform.OS.value == Platform.OS.Haiku
```
Comment (anmonteiro) [inline otherlibs/stdune/src/proc.ml:14]:
```suggestion
  if Sys.win32 || Platform.OS.value = Platform.OS.Haiku
```

File: vendor/spawn/src/spawn_stubs.c [hunk 2]
```diff
@@ -344,7 +344,6 @@ static void subprocess(int failure_fd, struct spawn_info *info)
   }
 
   pthread_sigmask(SIG_SETMASK, &info->child_sigmask, NULL);
-
```
Comment (anmonteiro) [inline vendor/spawn/src/spawn_stubs.c:347]: can you revert this whitespace diff?

**Review by anmonteiro — APPROVED** [review]

**Comment by anmonteiro** [comment]
@rgrinberg do you think this needs a changelog entry?

**Comment by rgrinberg** [comment]
I don't see why not. If this change is useful to anyone using Haiku, the change log entry would also be useful

**Comment by Alizter** [comment]
@Sylvain78 Thank you for the context. I recall now seeing something similar to do this when I had a look at Haiku support before. What the Haiku designers will do upstream shouldn't be something we rely on obviously, so the fix that you've given here seems sensible and rigid.

Please add a changelog entry together with links to the Haiku issue and this should be good to go.

**Pull request merged**